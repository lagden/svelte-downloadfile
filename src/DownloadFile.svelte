<script>
	export let props
	
	let {
		icon = false,
		type = 'INPUT',
		files = []
	} = props
	
	function getHrefFile(file) {
		if (file.type === 'BASE64') {
			return `data:${file.content_type};base64,${file.body}`
		}
		return file.file_url
	}
</script>

{#if type === 'INPUT'}
<div class="_input-downloadfile">
	<div class="_input-content-files">
		{#each files as file, i}
		<a href={getHrefFile(file)} download={file.file_name}>{file.file_name}</a>
		{/each}
	</div>
	{#if icon}
	<div class="_box-icon-downloadfile _icon-nocursor">
		<svg class="_icon-downloadfile">
			<use xlink:href="#{icon}" />
		</svg>
	</div>
	{/if}
</div>
{:else}
<a href={getHrefFile(files[0])} download={files[0].file_name} class="_btn-downloadfile" title={`Baixar ${files[0].file_name}`}>
	<svg class="_icon-downloadfile">
		<use xlink:href="#{icon}" />
	</svg>
</a>
{/if}

<style>
	:root {
		--magrini-downloadfile-background-color: hsl(240deg 18% 94% / 50%);
		--magrini-downloadfile-color: hsla(0, 0%, 0%, 0.7);
		--magrini-downloadfile-border: 1px solid;
		--magrini-downloadfile-border-color: hsl(240deg 8% 44% / 50%);
		--magrini-downloadfile-hover: hsl(210deg 50% 50%);
		--magrini-downloadfile-border-radius: 3px;
		--magrini-downloadfile-padding: 0.3em 1em 0.3em;
		--magrini-downloadfile-icon-size: 1.4em;
		--magrini-downloadfile-icon-color: hsl(240deg 8% 44%);
		--magrini-downloadfile-item-margin-right: 0.75em;
	}
	
	._input-downloadfile {
		background-color: var(--magrini-downloadfile-background-color);
		border: var(--magrini-downloadfile-border);
		border-color: var(--magrini-downloadfile-border-color);
		border-radius: var(--magrini-downloadfile-border-radius);
		padding: var(--magrini-downloadfile-padding);
		display: flex;
		align-items: center;
	}
	
	._btn-downloadfile {
		cursor: pointer;
		fill: var(--magrini-downloadfile-icon-color);
		color: var(--magrini-downloadfile-icon-color);
	}
	._btn-downloadfile:hover {
		fill: var(--magrini-downloadfile-hover);
		color: var(--magrini-downloadfile-hover);
	}
	
	._input-content-files {
		display: block;
		width: 100%;
	}
	
	._input-content-files a {
		display: inline-block;
		cursor: pointer;
		margin-right: var(--magrini-downloadfile-item-margin-right)
	}
	
	._icon-nocursor {
		cursor: default;
	}
	
	._icon-downloadfile {
		width: var(--magrini-downloadfile-icon-size);
		height: var(--magrini-downloadfile-icon-size);
	}
	
	._box-icon-downloadfile ._icon-downloadfile {
		fill: var(--magrini-downloadfile-icon-color);
		color: var(--magrini-downloadfile-icon-color);
	}
	
</style>
