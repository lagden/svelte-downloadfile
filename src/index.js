import DownloadFile from './DownloadFile.svelte'

export {
	DownloadFile
}
