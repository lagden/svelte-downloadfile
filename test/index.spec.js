/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import {DownloadFile} from '../src'

describe('DownloadFile', () => {
	afterEach(cleanup)

	test('should match snapshot default', () => {
		const {container} = render(DownloadFile)
		expect(container).toMatchSnapshot()
	})
})
