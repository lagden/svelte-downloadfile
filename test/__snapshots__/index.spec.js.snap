// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadFile should match snapshot default 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      display="none"
      focusable="false"
      height="0"
      version="1.1"
      width="0"
    >
      <symbol
        id="exportar_pdf"
        viewBox="0 0 32 32"
      >
        <path
          d="M23.5 32H6.1C3.9 32 2 30.1 2 27.9V4.1C2 1.9 3.9 0 6.1 0h17.4c2.3 0 4.1 1.9 4.1 4.1v4.4c0 .8-.6 1.4-1.4 1.4s-1.4-.6-1.4-1.4V4.1c0-.7-.6-1.3-1.3-1.3H6.1c-.7 0-1.3.6-1.3 1.3v23.7c0 .7.6 1.3 1.3 1.3h17.4c.7 0 1.3-.6 1.3-1.3 0-.8.6-1.4 1.4-1.4s1.4.6 1.4 1.4c.1 2.3-1.8 4.2-4.1 4.2z"
        />
        <path
          d="M14.5 14.7c-.3-.2-.6-.4-1.1-.6-.4-.1-.9-.2-1.5-.2H9.2v8.5h2v-2.8h.8c.5 0 1-.1 1.4-.2s.8-.3 1.1-.6c.3-.3.5-.6.6-.9.1-.4.2-.7.2-1.2 0-.4-.1-.8-.2-1.1-.1-.4-.3-.7-.6-.9zm-1.1 2c0 .2 0 .4-.1.5-.1.2-.1.3-.2.4s-.2.2-.4.2c-.2.1-.4.1-.6.1h-.9v-2.4h.8c.3 0 .5 0 .6.1.2.1.3.1.4.2s.2.2.2.3c.2.3.2.4.2.6zM22.4 15.1c-.4-.4-.8-.7-1.4-.9-.5-.2-1.1-.3-1.8-.3H16v8.5h3.3c.6 0 1.2-.1 1.8-.3.5-.2 1-.5 1.4-.9s.7-.8.9-1.4c.2-.5.3-1.1.3-1.7s-.1-1.2-.3-1.7c-.3-.5-.6-.9-1-1.3zm-.8 3.1c0 .4-.1.8-.2 1.1s-.3.6-.5.8c-.2.2-.4.4-.7.5s-.6.2-1 .2H18v-5.1h1.3c.4 0 .7.1 1 .2s.5.3.7.5c.2.2.4.5.5.8.1.2.1.6.1 1zM30 15.6v-1.7h-5.6v8.5h2v-3.2h3v-1.7h-3v-1.9z"
        />
      </symbol>
    </svg>
     
    <div
      class="__downloadfile"
    >
       
      <div
        class="_input-base"
      >
        <div
          class="_input-content-files"
        />
         
      </div>
    </div>
  </div>
</body>
`;
